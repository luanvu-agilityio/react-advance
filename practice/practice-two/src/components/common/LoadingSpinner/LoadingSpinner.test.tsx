import { render, screen } from '@testing-library/react'
import { LoadingSpinner } from './index'

// Mock the Radix UI Spinner component
jest.mock('@radix-ui/themes', () => ({
  Spinner: ({ size }: { size: string }) => (
    <div data-testid="spinner" data-size={size}>
      Spinner
    </div>
  ),
  Theme: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="theme-wrapper">{children}</div>
  ),
}))

// Mock the Text component
jest.mock('../Text/index', () => ({
  __esModule: true,
  default: ({ text, as }: { text: string; as: string }) => (
    <div data-testid="text-component" data-as={as}>
      {text}
    </div>
  ),
}))

describe('LoadingSpinner', () => {
  it('renders without crashing', () => {
    render(<LoadingSpinner />)

    expect(screen.getByTestId('spinner')).toBeInTheDocument()
  })

  it('displays the loading text', () => {
    render(<LoadingSpinner />)

    expect(screen.getByText('Loading product details...')).toBeInTheDocument()
  })

  it('renders the spinner with correct size', () => {
    render(<LoadingSpinner />)

    const spinner = screen.getByTestId('spinner')
    expect(spinner).toHaveAttribute('data-size', '3')
  })

  it('renders text as paragraph element', () => {
    render(<LoadingSpinner />)

    const textComponent = screen.getByTestId('text-component')
    expect(textComponent).toHaveAttribute('data-as', 'p')
  })

  it('wraps components in Theme provider', () => {
    render(<LoadingSpinner />)

    expect(screen.getByTestId('theme-wrapper')).toBeInTheDocument()
  })

  it('has proper container structure', () => {
    const { container } = render(<LoadingSpinner />)

    // Check if LoadingContainer is the root element
    const loadingContainer = container.firstChild
    expect(loadingContainer).toBeInTheDocument()

    // Check if SpinnerWrapper is inside LoadingContainer
    const spinnerWrapper = screen.getByTestId('theme-wrapper').parentElement
    expect(spinnerWrapper).toBeInTheDocument()
  })

  it('renders all required components', () => {
    render(<LoadingSpinner />)

    // Check all components are rendered
    expect(screen.getByTestId('spinner')).toBeInTheDocument()
    expect(screen.getByTestId('text-component')).toBeInTheDocument()
    expect(screen.getByTestId('theme-wrapper')).toBeInTheDocument()
    expect(screen.getByText('Loading product details...')).toBeInTheDocument()
  })
})
