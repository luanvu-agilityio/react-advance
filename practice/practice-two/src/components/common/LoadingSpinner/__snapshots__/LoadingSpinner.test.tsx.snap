// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner matches snapshot 1`] = `
<div>
  <div
    class="sc-Qotzb eMIPCM"
    style="min-height: 400px;"
  >
    <div
      class="sc-fYsHOw cgGqeq"
    >
      <div
        data-size="3"
        data-testid="spinner"
      >
        Spinner
      </div>
      <div
        data-as="p"
        data-testid="text-component"
      >
        Loading product details...
      </div>
    </div>
  </div>
</div>
`;

exports[`LoadingSpinner matches snapshot with custom height 1`] = `
<div>
  <div
    class="sc-Qotzb eMIPCM"
    style="min-height: 200px;"
  >
    <div
      class="sc-fYsHOw cgGqeq"
    >
      <div
        data-size="3"
        data-testid="spinner"
      >
        Spinner
      </div>
      <div
        data-as="p"
        data-testid="text-component"
      >
        Loading product details...
      </div>
    </div>
  </div>
</div>
`;

exports[`LoadingSpinner matches snapshot with custom message 1`] = `
<div>
  <div
    class="sc-Qotzb eMIPCM"
    style="min-height: 400px;"
  >
    <div
      class="sc-fYsHOw cgGqeq"
    >
      <div
        data-size="3"
        data-testid="spinner"
      >
        Spinner
      </div>
      <div
        data-as="p"
        data-testid="text-component"
      >
        Custom loading message
      </div>
    </div>
  </div>
</div>
`;

exports[`LoadingSpinner matches snapshot with different size 1`] = `
<div>
  <div
    class="sc-Qotzb eMIPCM"
    style="min-height: 400px;"
  >
    <div
      class="sc-fYsHOw cgGqeq"
    >
      <div
        data-size="2"
        data-testid="spinner"
      >
        Spinner
      </div>
      <div
        data-as="p"
        data-testid="text-component"
      >
        Loading product details...
      </div>
    </div>
  </div>
</div>
`;
